<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>KisAutoLevelsWidget</class>
 <widget class="QWidget" name="KisAutoLevelsWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>300</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>10</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string comment="Contrast enhancement section in auto levels">Shadows and Highlights</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
      </property>
      <property name="horizontalSpacing">
       <number>5</number>
      </property>
      <property name="verticalSpacing">
       <number>5</number>
      </property>
      <property name="leftMargin">
       <number>10</number>
      </property>
      <property name="topMargin">
       <number>10</number>
      </property>
      <property name="rightMargin">
       <number>10</number>
      </property>
      <property name="bottomMargin">
       <number>10</number>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string comment="Contrast enhancement method used in auto levels">Method:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="comboBoxShadowsAndLightsMethod">
        <item>
         <property name="text">
          <string comment="Use same black and white points for all the channels in auto levels">Adjust monochromatic contrast</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string comment="Use different black and white points for each channel in auto levels">Adjust per channel contrast</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string comment="Percentage of the histogram to clip in the shadous side in auto levels">Shadows clipping:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string comment="Percentage of the histogram to clip in the highlights side in auto levels">Highlights clipping:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string comment="Maximum distance the black/white points can be moved from their relative extremes in auto levels">Maximum offset:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="KisDoubleSliderSpinBox" name="sliderShadowsAndLightsMaximumOffset" native="true"/>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string comment="Output color used for the shadows in auto levels">Shadows color:</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string comment="Output color used for the highlights in auto levels">Highlights color:</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="KisColorButton" name="buttonShadowsColor" native="true"/>
      </item>
      <item row="5" column="1">
       <widget class="KisColorButton" name="buttonHighlightsColor" native="true"/>
      </item>
      <item row="1" column="1">
       <widget class="KisDoubleSliderSpinBox" name="sliderShadowsClipping" native="true"/>
      </item>
      <item row="2" column="1">
       <widget class="KisDoubleSliderSpinBox" name="sliderHighlightsClipping" native="true"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string comment="Midtones enhancement section in auto levels">Midtones</string>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
      </property>
      <property name="horizontalSpacing">
       <number>5</number>
      </property>
      <property name="verticalSpacing">
       <number>5</number>
      </property>
      <property name="leftMargin">
       <number>10</number>
      </property>
      <property name="topMargin">
       <number>10</number>
      </property>
      <property name="rightMargin">
       <number>10</number>
      </property>
      <property name="bottomMargin">
       <number>10</number>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string comment="Midtones enhancement method used in auto levels">Method:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="comboBoxMidtonesMethod">
        <item>
         <property name="text">
          <string comment="Do not perform midtones enhancement in auto levels">Don't adjust the midtones</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string comment="Use the median point of the histogram as midtone point in auto levels">Use the median to adjust the midtones</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string comment="Use the mean value of the histogram as midtone point in auto levels">Use the mean value to adjust the midtones</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string comment="Output color used for the midtones in auto levels">Color:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="KisColorButton" name="buttonMidtonesColor" native="true"/>
      </item>
      <item row="1" column="1">
       <widget class="KisDoubleSliderSpinBox" name="sliderMidtonesAdjustmentAmount" native="true"/>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string comment="How much the midtone point used in auto levels is close to the center of the histogram (0%) or to the point selected in the method (100%)">Amount:</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KisDoubleSliderSpinBox</class>
   <extends>QWidget</extends>
   <header location="global">kis_slider_spin_box.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>KisColorButton</class>
   <extends>QWidget</extends>
   <header location="global">kis_color_button.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
